### PR TITLE
Building Student Dashboard Component

### PR DESCRIPTION
**Pull request by peterdavidconley**
## Description

Creating the Student Dashboard component where student can access project resources and stories from current teacher. Based on Figma design: https://www.figma.com/file/VoV0f8wvr8t3GDz8q4ffU7/Main?node-id=2263%3A21718.

Co-authored-by: Peter Conley <[peter.david.conley@gmail.com](mailto:peter.david.conley@gmail.com)>
Co-authored-by: Hannah Brog <[bugsbrog@gmail.com](mailto:bugsbrog@gmail.com)>

Fixes # (issue)

## Loom Video

Loom link: [https://www.loom.com/share/a9f4699bac7b4f839fd155be264449f3](https://www.loom.com/share/a9f4699bac7b4f839fd155be264449f3)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have removed unnecessary comments/console logs from my code
- [x] I have made corresponding changes to the documentation if necessary (optional)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] No duplicate code left within changed files
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

**Review by dconley1212 — APPROVED** [review]
The structure of this component looks great and it looks like you have been incorporating ant design elements to follow the design of the other web pages on the coderheroes site. Great work!

**Review by magoha01 — APPROVED** [review]
Really nice job, this is coming along nicely! I can tell you are putting a lot of thought into the structure of this flow. 

Now that you have the layout down, I think it would be a really good to continue to implement the antd components throughout your design in lieu of the divs (ex. creating a student card). I also think it would be good to go back and implement the preset themes for your colors/buttons/headers/fonts/etc where possible in your LESS file. 

Keep up the good work!

**Review by sharontsanchez — APPROVED** [review]
Good job implementing the student dashboard component. The approach of one flow at a time helps with both Figma designing and frontend development. Good job with breaking each component into sub section inside of Student Home. and connecting and creating a path to the student route inside of index.js Great to see Ant Design as well. A good start to the student flow.

**Review by joeys1122 — APPROVED** [review]
Very nice work Hannah and Peter! The styles and overall structure of the components look very clean, this code provides a great starting point for further additions to the student flow!

**Review by JessWillCode — COMMENTED** [review]
Left my comments in the conversation tab. I can't wait to see more!

File: src/styles/StudentStyles/index.less [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+.current-student-db img{
+    margin: 5% 0%;
+    width: 50%;
+    height: 50%;
+    border-radius: 100%;
+}
+
+.current-student-db h1{
+    color: white;
+    text-transform: uppercase;
```
[the remark this answers is not in the document]
Comment (nyamekyeannor) [reply src/styles/StudentStyles/index.less:43]: I agree with Jess! Amazing job using text-transform. I've seen many take the shortcut route and "hardcode uppercase" in the code.

File: src/components/pages/StudentHome/StudentSidebar.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      theme="light"
+      collapsible
+      collapsed={collapsed}
+      onCollapse={onCollapse}
+    >
+      <Menu
+        className="parent-dashboard-sidebar"
+        defaultSelectedKeys={['2']}
+        defaultOpenKeys={['sub1']}
+        style={{ height: '100vh' }}
```
[the remark this answers is not in the document]
Comment (nyamekyeannor) [reply src/components/pages/StudentHome/StudentSidebar.js:43]: Perhaps your styles were being overridden? If so you could use "!important" kindly take a look at this site. w3schools.com/css/css_important.asp

**Review by nyamekyeannor — COMMENTED** [review]
Amazing job Peter! This is detailed work. I love the style and the way you’ve structured your components. I am curious however to know why you decided to use in-line styles on line 43 of StudentSidebar.JS? I’m guessing your styles were being overridden? Kindly check out the link i left below line 43 to see if using "!important" could assist in the future. Amazing work man.

**Pull request merged**